packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -369,96 +460,111 @@ contract Colosseum is Initializable, Semver {
 
         require(
             !verifiedPublicInputs[publicInputHash],
-            "Colosseum: public input that has already been validated cannot be used again."
+            "Colosseum: public input that has already been validated cannot be used again"
         );
 
         // TODO(pangssu): waiting for the new Verifier.sol to complete.
-        // require(ZK_VERIFIER.verify(_zkproof, _pair, publicInputHash), "Colosseum: invalid proof");
+        // require(ZK_VERIFIER.verify(_zkproof, _pair), "Colosseum: invalid proof");
+        emit Proven(_outputIndex, msg.sender, block.timestamp);
+
+        // Scope to call the security council, to avoid stack too deep.
+        {
+            Types.CheckpointOutput memory output = L2_ORACLE.getL2Output(_outputIndex);
+
+            bytes memory callbackData = abi.encodeWithSelector(
+                this.dismissChallenge.selector,
+                _outputIndex,
+                msg.sender,
+                challenge.asserter,
+                output.outputRoot
+            );
+
+            // Request outputRoot validation to security council
+            SecurityCouncil(SECURITY_COUNCIL).requestValidation(
+                output.outputRoot,
+                output.l2BlockNumber,
+                callbackData
+            );
+        }
 
         verifiedPublicInputs[publicInputHash] = true;
-        challenge.outputRoot = _outputRoot;
+        delete challenges[_outputIndex][msg.sender];
 
-        _callSecurityCouncil(_outputIndex, _outputRoot);
+        // Delete output root.
+        L2_ORACLE.replaceL2Output(_outputIndex, DELETED_OUTPUT_ROOT, msg.sender);
+        // The challenger's bond is also included in the bond for that output.
+        L2_ORACLE.VALIDATOR_POOL().increaseBond(_outputIndex, msg.sender);
     }
 
     /**
-     * @notice Deletes the challenge because the challenger timed out.
-     *         The winner is the asserter.
+     * @notice Calls a private function that deletes the challenge because the challenger has timed out.
+     *         Reverts if the challenger hasn't timed out.
      *
      * @param _outputIndex Index of the L2 checkpoint output.
+     * @param _challenger  Address of the challenger.
      */
-    function challengerTimeout(uint256 _outputIndex) external {
-        Types.Challenge storage challenge = challenges[_outputIndex];
-        _validateTurn(challenge);
+    function challengerTimeout(uint256 _outputIndex, address _challenger)
+        external
+        outputNotFinalized(_outputIndex)
```

Comment:
I think it is necessary because when output is finalized, the bond is unbonded. After that, `_challengerTimeout` moves the bond to the already unbonded bond which may never be unbonded again.
But I agree that we need a way to handle the pending bonds of timed out challenger after output is finalized. The pending bonds would be locked forever.